tools/attach_logs_in_jira.py

```diff
@@ -0,0 +1,98 @@
+import argparse
+import logging
+import re
+import shutil
+import tarfile
+from pathlib import Path
+from functools import partial
+
+import jira
+
+from consts import JIRA_PROJECT, JIRA_SUMMARY, JIRA_SERVER
+
+JIRA_ATTACHMENT_MAX_SIZE = 50 * 2**20  # 50MB
+
+log = logging.getLogger(__name__)
+
+
+def _get_or_create_jira_ticket(jira_client: jira.JIRA, summary: str) -> jira.Issue:
+    matching_issues = jira_client.search_issues(f'project = "{JIRA_PROJECT}" AND summary = {summary}')
+
+    if len(matching_issues) == 0:
+        log.info("Creating a placeholder issue for %s", summary)
+        return jira_client.create_issue(
+            project=JIRA_PROJECT,
+            summary=summary,
+            description=(
+                "Logs are still being uploaded.\nTicket description and signatures will get processed shortly..."
+            ),
+        )
+
+    if len(matching_issues) == 1:
+        log.debug("Using pre-existing issue %s", matching_issues[0])
+        return matching_issues[0]
+
+    raise RuntimeError(f"Found more than one matching issue for {cluster_logs}: {' '.join(matching_issues)}")
+
+
+def _archive_dir(directory: Path) -> Path:
+    archived_file = Path(f"{directory}.tar.gz")
+    log.debug("Archiving directory %s to %s", directory, archived_file)
+    with tarfile.open(archived_file, "w:gz") as tar:
+        tar.add(directory, directory.name)
+
+    return archived_file
+
+
+def _split_file(path: Path, chunk: int) -> None:
+    log.debug("Splitting %s to several files", path)
+    with path.open("rb") as whole_file:
+        read_chunk = partial(whole_file.read, chunk)
+        for i, chunk in enumerate(iter(read_chunk, "")):
+            with open(f"{path}-part-{i}", "wb") as part:
+                part.write(chunk)
+
+
+def attach_logs_in_jira(source_dir: Path, jira_access_token: str):
+    jira_client = jira.JIRA(JIRA_SERVER, token_auth=jira_access_token, validate=True)
+
+    for cluster_logs in source_dir.iterdir():
+        issue = _get_or_create_jira_ticket(jira_client, JIRA_SUMMARY.format(cluster_logs))
+
+        for artifact in cluster_logs.iterdir():
+            if artifact.is_dir():
+                archived_file = _archive_dir(artifact)
+                shutil.rmtree(artifact)
+                artifact = archived_file
+
+            if artifact.stat().st_size > JIRA_ATTACHMENT_MAX_SIZE:
+                _split_file(artifact, chunk=JIRA_ATTACHMENT_MAX_SIZE)
+                artifact.unlink()
+
+        for artifact in cluster_logs.iterdir():
+            if any(attachment.name == artifact.name for attachment in issue.fields.attachment):
```

Comment:
Possibly be more readable with `in`:

```suggestion
            if artifact.name in (attachment.name for attachment in issue.fields.attachment):
```